File: internal/app/events.go
package app

import (
	"fmt"
	"time"

	"github.com/gin-gonic/gin"
	socketio "github.com/googollee/go-socket.io"
)

// InitializeSocketEvents sets up the Socket.IO event handlers
func InitializeSocketEvents(server *socketio.Server) {
	server.OnEvent("/chat", "joined", joined)
	server.OnEvent("/chat", "text", text)
	server.OnEvent("/chat", "left", left)
}

// joined handles the 'joined' event when a client enters a room
func joined(s socketio.Conn, msg string) {
	room := s.Context().(string)
	s.Join(room)
	s.Emit("status", gin.H{
		"msg": fmt.Sprintf("%s has entered the room.", s.Context().(string)),
	})
}

// text handles the 'text' event when a client sends a new message
func text(s socketio.Conn, msg string) {
	room := s.Context().(string)
	s.To(room).Emit("message", gin.H{
		"msg": fmt.Sprintf("%s: %s", s.Context().(string), msg),
	})
}

// left handles the 'left' event when a client leaves a room
func left(s socketio.Conn, msg string) {
	room := s.Context().(string)
	s.Leave(room)
	s.To(room).Emit("status", gin.H{
		"msg": fmt.Sprintf("%s has left the room.", s.Context().(string)),
	})
}

File: Flask-SocketIO-Chat-7d8f9/app/main/events.py
from time import time
from flask import session
from flask_socketio import emit, join_room, leave_room
from .. import socketio
from flasgger import swag_from

@socketio.on('joined', namespace='/chat')
@swag_from({
    'tags': ['chat'],
    'description': 'Sent by clients when they enter a room. A status message is broadcast to all people in the room.',
    'parameters': [
        {
            'name': 'message',
            'in': 'body',
            'description': 'Message data',
            'required': True,
            'schema': {
                'type': 'object',
                'properties': {
                    'room': {'type': 'string'}
                }
            }
        }
    ],
    'responses': {
        '200': {
            'description': 'Status message broadcasted successfully'
        }
    }
})
def joined(message):
    """
    Sent by clients when they enter a room.
    A status message is broadcast to all people in the room.
    """
    room = session.get('room')
    join_room(room)
    emit('status', {
        'msg': f"{session.get('name')} has entered the room.",
    }, to=room)

@socketio.on('text', namespace='/chat')
@swag_from({
    'tags': ['chat'],
    'description': 'Sent by a client when the user entered a new message. The message is sent to all people in the room.',
    'parameters': [
        {
            'name': 'message',
            'in': 'body',
            'description': 'Message data',
            'required': True,
            'schema': {
                'type': 'object',
                'properties': {
                    'msg': {'type': 'string'}
                }
            }
        }
    ],
    'responses': {
        '200': {
            'description': 'Message broadcasted successfully'
        }
    }
})
def text(message):
    """
    Sent by a client when the user entered a new message.
    The message is sent to all people in the room.
    """
    room = session.get('room')
    emit('message', {
        'msg': f"{session.get('name')}: {message['msg']}",
    }, to=room)

@socketio.on('left', namespace='/chat')
@swag_from({
    'tags': ['chat'],
    'description': 'Sent by clients when they leave a room. A status message is broadcast to all people in the room.',
    'parameters': [
        {
            'name': 'message',
            'in': 'body',
            'description': 'Message data',
            'required': True,
            'schema': {
                'type': 'object',
                'properties': {
                    'room': {'type': 'string'}
                }
            }
        }
    ],
    'responses': {
        '200': {
            'description': 'Status message broadcasted successfully'
        }
    }
})
def left(message):
    """
    Sent by clients when they leave a room.
    A status message is broadcast to all people in the room.
    """
    room = session.get('room')
    leave_room(room)
    emit('status', {
        'msg': f"{session.get('name')} has left the room.",
    }, to=room)